src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -943,41 +939,37 @@ def _get_default_address_pool(cli_ctx, resource_group, balancer_name, balancer_t
     return values[0]
 
 
+def _validate_vmss_single_placement_group(namespace):
+    if namespace.platform_fault_domain_count is not None and namespace.zones is None:
+        raise CLIError('usage error: --platform-fault-domain-count COUNT --zones ZONES')
```

Comment:
@gatneil, I am adding this client side validation to ensure users can only set `platformFaultDomainCount` on zonal scale-sets, as the error from service side is rather vague. However if CRP is getting it work with non-zonal ones soon, then I don't think we need such validation as it would end up blocking people. Thoughts?